apps/site/pages/docs/api-extensions/extending-api-schema.mdx

```diff
@@ -0,0 +1,378 @@
+This field can store custom information related to a product in your store. For example, you might use this field to store additional product details that are not present in the native FastStore API schema. 
+
+```graphql filename="graphql/vtex/typeDefs/product.graphql"
+
+extend type StoreProduct {
+  """
+  Custom data extending StoreProduct
+  """
+  customData: String!
+}
+
+```
+
+<Callout type="warning" emoji="⚠️">
+  The typeDefs files must have `.graphql` extension.
+  Also, the `index.ts` is not needed for typeDefs, only for resolvers.
+</Callout>
+
+Now, refer to the next step to create the resolvers to provide the actual data to the `customData` field. 
+
+#### Step 3 - Creating resolvers
+
+Now, you'll create a resolver to define how the new `customData` field should be resolved, or in other words, what data or logic should be associated with them when queried.
+
+1. Create a new `<resolver>.graphql` file inside the `vtex/resolvers` folder. Let's continue the [StoreProduct](https://v1.faststore.dev/reference/api/objects#storeproduct) type
+example, so the name of the file is `product.ts`.
+
+<Callout type="info" emoji="ℹ️">
+  The way that it will be organized inside this folder is flexible, it's possible to create multiple resolvers files or group them all together in just one file.
+</Callout>
+
+2. Define the resolver in the `product.ts` file. This resolver specifies how to fetch the data for the `customData` field based on the root object, which represents the `StoreProduct`.
+
+```ts copy filename="graphql/vtex/resolvers/product.ts"
+
+import type { StoreProductRoot } from '@faststore/api'
+
+const productResolver = {
+  StoreProduct: {
+    customData: (root: StoreProductRoot) => {
+      return 'My item id: ' + root.itemId
+    },
+  },
+}
+
+export default productResolver
+```
+
+Note that you can leverage the use of TypeScript by typing the root param as [StoreProductRoot](https://v1.faststore.dev/reference/api/root-object-fields#root-product):
+
+![using-root-field-type](https://vtexhelp.vtexassets.com/assets/docs/src/root-field-ts___3a10ec8a1aeba9770c7ee982995d55bc.gif)
+
+3. Open the `index.ts` file inside the `vtex/resolvers` folder and import the resolver, `StoreProductResolver`, you created in the previous step. 
+
+4. Create a `resolvers` object that includes the imported resolver(s).
+
+```ts copy filename="graphql/vtex/resolvers/index.ts"
+
+import { default as StoreProductResolver } from './product'
+
+const resolvers = {
+  ...StoreProductResolver,
+}
+
+export default resolvers
+```
+
+Once you have defined these files in the `vtex` folder, the new fields are already available to use. 
+
+<Callout type="warning" emoji="⚠️">
+  If the changes you made still not available, refer to the troubleshooting section [GraphQL changes not visible during development](/docs/api-extension/troubleshooting#graphql-changes-not-visible-during-development).
+</Callout>
+ 
+
+## Extending third-party API schemas
```

Comment:
```suggestion
## Extending FastStore API with third-party API schemas
```

The third-party API schemas don't exist until the user defines it. We are actually extending FastStore API (in this case, always adding, not modifying its behavior) with third-party API schemas.